File: fix_styles.js
const fs = require('fs');

// 1. Fix login.html
let login = fs.readFileSync('login.html', 'utf8');
login = login.replace(
  'body { display: flex; justify-content: center; align-items: center; min-height: 100vh; background: var(--bg-color); margin: 0; padding: 20px; box-sizing: border-box; }',
  'body { background: var(--bg-color); margin: 0; padding: 0; box-sizing: border-box; }'
);
login = login.replace(
  '.auth-card input { width: 100%;',
  '.auth-card input:not([type="checkbox"]) { width: 100%;'
);

if (login.indexOf('<div class="auth-wrapper">') === -1) {
  login = login.replace('<div class="auth-card">', '<div class="auth-wrapper">\n  <div class="auth-card">');
  login = login.replace('</div>\n\n  <div id="toast-container"></div>', '  </div>\n</div>\n\n  <div id="toast-container"></div>');
}
fs.writeFileSync('login.html', login);

// 2. Fix style.css for retailer/customer register
let style = fs.readFileSync('style.css', 'utf8');
style = style.replace(
  '.auth-container input, .auth-container select {',
  '.auth-container input:not([type="checkbox"]), .auth-container select {'
);
fs.writeFileSync('style.css', style);

console.log('Fixed styles successfully.');

File: scripts/fix_styles.py
import re
import os

files = ['frontend/login.html', 'frontend/customer-register.html', 'frontend/retailer-register.html']

for f in files:
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
    
    # Update body style
    content = re.sub(r'<body[^>]*>', '<body style="display: flex; flex-direction: column; min-height: 100dvh; margin: 0; padding: 0; box-sizing: border-box;">', content)
    
    # Cache bust css
    content = re.sub(r'href="style\.css(\?v=\d+)?"', 'href="style.css?v=6"', content)
    
    with open(f, 'w', encoding='utf-8') as file:
        file.write(content)

# Update style.css
css_path = 'frontend/style.css'
with open(css_path, 'r', encoding='utf-8') as file:
    css = file.read()

# Remove old body:has(.auth-wrapper)
css = re.sub(r'body:has\(\.auth-wrapper\) \{[\s\S]*?body:has\(\.auth-wrapper\) footer \{[\s\S]*?\}\s*', '', css)

# Update auth-wrapper and footer
auth_wrapper_regex = r'\.auth-wrapper \{[\s\S]*?box-sizing: border-box;\n\}'
replacement = '''.auth-wrapper {
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
  background: linear-gradient(135deg, rgba(255,106,0,0.05) 0%, rgba(255,255,255,0) 100%);
  padding: 20px;
  box-sizing: border-box;
}

.auth-wrapper ~ footer {
  margin-top: auto !important;
}'''

css = re.sub(auth_wrapper_regex, replacement, css)

with open(css_path, 'w', encoding='utf-8') as file:
    file.write(css)

print("Updates completed successfully.")
